frontend/src/LandingPage.js

```diff
@@ -1,50 +1,269 @@
-import React from 'react';
+import React, { useState, useEffect, useRef } from "react";
 import './App.css';
 import { Client, Status } from "@googlemaps/google-maps-services-js";
 import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
 import { Link } from 'react-router-dom';
 import { createBrowserHistory } from "history";
 import SECRET_KEY from './config';
+import ITINERARY_KEY from './configMap';
 import Map from './Itinerary';
 import Button from "react-bootstrap/Button";
 import ReactDOM from 'react-dom';
 import ImgKey from './configImg';
+// import DayPickerInput from 'react-day-picker/DayPickerInput';
+// import 'react-day-picker/lib/style.css';
+
+let autoComplete1, autoComplete2, currentUserId;
 
 function LandingPage() {
-    function createListElem(attraction){
+    let google, map, markers = [];
+    function initMap() {
+        google = window.google;
+
+        const cityLocation = {
+          lat: 48.1351,
+          lng: 11.5820
+        };
+        map = new google.maps.Map(document.getElementById("map"), {
+          zoom: 11,
+          center: cityLocation
+        });
+    }
+    useEffect(() => {
+        loadScript(
+            'https://maps.googleapis.com/maps/api/js?key=' + Object.values(ITINERARY_KEY)[0] + '&libraries=places'
+        );
+        authentication();
+        if (!window.google) {
```

Comment:
Could you please clarify, what's going on here? In which cases do you append a script to the page? Isn't it possible to make it a react component instead of pure javascript embedded into html?

Reply:
useEffect is very similar to onComponentMount function. We need to append that script before compiling script. We couldn't find a way to make it a react component.

Reply:
If we do not make sure that we use the map after it is correctly loaded, it will not be recognized. useEffect makes sure that everthing is loaded (it combines componentDidMount, componentDidUpdate, and componentWillUnmount)